src/full_energy.c

```diff
@@ -206,26 +237,179 @@ int  full_energy(int                  gridcell,
+        // Handle crop tiles (fallow and crop sub-tiles)
+        if (options.CROPFRAC && veg_con[iveg].crop_frac_active) {
+          for (cridx=veg_con[iveg].crop_frac_idx; cridx<veg_con[iveg].crop_frac_idx+2; cridx++) {
+            if (cridx % 2 == 0) { //fallow part
+              all_vars_crop->veg_var[cridx][band].vegcover = MIN_VEGCOVER;
+              all_vars_crop->veg_var[cridx][band].albedo = BARE_SOIL_ALBEDO;
+              all_vars_crop->veg_var[cridx][band].LAI = 0;
+            }
+            else { //crop part
+              // Convert LAI from global to local
+	      //veg_var[iveg][band].vegcover=1;
+              all_vars_crop->veg_var[cridx][band].vegcover = veg_var[iveg][band].vegcover/veg_var[iveg][band].crop_frac;
+              if (all_vars_crop->veg_var[cridx][band].vegcover > 1) all_vars_crop->veg_var[cridx][band].vegcover = 1;
+              if (all_vars_crop->veg_var[cridx][band].vegcover < MIN_VEGCOVER) all_vars_crop->veg_var[cridx][band].vegcover = MIN_VEGCOVER;
+              all_vars_crop->veg_var[cridx][band].albedo = (veg_var[iveg][band].albedo - (1-veg_var[iveg][band].crop_frac)*BARE_SOIL_ALBEDO)/veg_var[iveg][band].crop_frac;
+              if (all_vars_crop->veg_var[cridx][band].albedo < H2O_SURF_ALBEDO) all_vars_crop->veg_var[cridx][band].albedo = H2O_SURF_ALBEDO;
+              all_vars_crop->veg_var[cridx][band].LAI = veg_var[iveg][band].LAI/(all_vars_crop->veg_var[cridx][band].vegcover*veg_var[iveg][band].crop_frac);
+              all_vars_crop->veg_var[cridx][band].Wdew = veg_var[iveg][band].Wdew/(all_vars_crop->veg_var[cridx][band].vegcover*veg_var[iveg][band].crop_frac);
+              all_vars_crop->veg_var[cridx][band].Wdmax = all_vars_crop->veg_var[cridx][band].LAI*LAI_WATER_FACTOR;
+              all_vars_crop->snow[cridx][band].snow_canopy = snow[iveg][band].snow_canopy/(all_vars_crop->veg_var[cridx][band].vegcover*veg_var[iveg][band].crop_frac);
+            }
+          }
+        }
+        else {
+          // Convert LAI from global to local
+          veg_var[iveg][band].LAI /= veg_var[iveg][band].vegcover;
+          veg_var[iveg][band].Wdew /= veg_var[iveg][band].vegcover;
+          veg_var[iveg][band].Wdmax = veg_var[iveg][band].LAI*LAI_WATER_FACTOR;
+          snow[iveg][band].snow_canopy /= veg_var[iveg][band].vegcover;
+        }
       }
     }
   }
 
+  /***************************************
+    Irrigation demand.
+    Based on upper layer soil moisture.
+  ******************************************/
+  irrig = 0;
+  if (options.IRRIGATION) {
+
+    if (rec >= 0) {
+      for(iveg = 0; iveg < Nveg; iveg++){
+        for ( band = 0; band < Nbands; band++ ) {
+	  cell[iveg][band].irr_extract=0; //ingjerd added, for initializing. needed, or maybe not?
+	  veg_class = veg_con[iveg].veg_class; //ingjerd added
+	  veg_var[iveg][band].irrig=0.; //ingjerd added for initialization. hm... not sure about this!
+	  if (options.CROPFRAC  && veg_con[iveg].crop_frac_active)
```

Comment:
@tbohn I want to confirm about your comment here. Correct me if I'm wrong. To me the if statement (L345) only initialize irrig and it does not prevent computing irrigation if CROPFRAC is false or crop_frac_active is false. So your suggestion is just removing this line (L345) and always initializing irrig?